pkg/runtime/runtime_test.go

```diff
@@ -4117,3 +4122,132 @@ func TestComponentsCallback(t *testing.T) {
 
 	assert.True(t, callbackInvoked, "component callback was not invoked")
 }
+
+func TestGRPCProxy(t *testing.T) {
+	// setup gRPC server
+	serverPort, _ := freeport.GetFreePort()
+	teardown, err := runGRPCApp(serverPort)
+	require.NoError(t, err)
+	defer teardown()
+
+	mockNameResolution := nr_loader.NameResolution{
+		Names: []string{"mdns"}, // for standalone mode
+		FactoryMethod: func() nameresolution.Resolver {
+			mockResolver := new(daprt.MockResolver)
+			// proxy to server anytime
+			mockResolver.On("Init", mock.Anything).Return(nil)
+			mockResolver.On("ResolveID", mock.Anything).Return(fmt.Sprintf("localhost:%d", serverPort), nil)
+			return mockResolver
+		},
+	}
+
+	// setup proxy
+	rt := NewTestDaprRuntimeWithProtocol(modes.StandaloneMode, "grpc", serverPort)
+	internalPort, _ := freeport.GetFreePort()
+	rt.runtimeConfig.InternalGRPCPort = internalPort
+	defer stopRuntime(t, rt)
+
+	go func() {
+		rt.Run(WithNameResolutions(mockNameResolution))
+	}()
+	defer rt.Shutdown(0)
+
+	time.Sleep(time.Second)
+
+	req := &pb.PingRequest{Value: "foo"}
+
+	t.Run("proxy single streaming request", func(t *testing.T) {
+		ctx, cancel := context.WithTimeout(context.TODO(), time.Second)
+		defer cancel()
+		stream, err := pingStreamClient(ctx, internalPort)
+		require.NoError(t, err)
+
+		require.NoError(t, stream.Send(req), "sending to PingStream must not fail")
+		resp, err := stream.Recv()
+		require.NoError(t, err)
+		require.NotNil(t, resp, "resp must not be nil")
+
+		require.NoError(t, stream.CloseSend(), "no error on close send")
+	})
+
+	t.Run("proxy concurrent streaming requests", func(t *testing.T) {
+		ctx1, cancel := context.WithTimeout(context.TODO(), time.Second)
+		defer cancel()
+		stream1, err := pingStreamClient(ctx1, internalPort)
+		require.NoError(t, err)
+
+		ctx2, cancel := context.WithTimeout(context.TODO(), time.Second)
+		defer cancel()
+		stream2, err := pingStreamClient(ctx2, internalPort)
+		require.NoError(t, err)
+
+		require.NoError(t, stream1.Send(req), "sending to PingStream must not fail")
+		resp, err := stream1.Recv()
+		require.NoError(t, err)
+		require.NotNil(t, resp, "resp must not be nil")
+
+		require.NoError(t, stream2.Send(req), "sending to PingStream must not fail")
+		resp, err = stream2.Recv()
+		require.NoError(t, err)
+		require.NotNil(t, resp, "resp must not be nil")
+
+		require.NoError(t, stream1.CloseSend(), "no error on close send")
+		require.NoError(t, stream2.CloseSend(), "no error on close send")
+	})
```

Comment:
I made a concurrent request unittest here because the issue is caused by the combination of `GetGRPCConnection` and `TransparentHandler`.